src/server/terrain/structure_building_blocks.zig

```diff
@@ -101,6 +102,100 @@ pub fn isOriginBlock(block: Block) bool {
 	return block.typ == originBlockNumericId;
 }
 
+pub const StructureBuildingBlock = struct {
+	children: []AliasTable(Child),
+	blueprints: *[4]BlueprintEntry,
+
+	fn initFromZon(stringId: []const u8, zon: ZonElement) !StructureBuildingBlock {
+		const blueprintId = zon.get(?[]const u8, "blueprint", null) orelse {
+			std.log.err("['{s}'] Missing blueprint field.", .{stringId});
+			return error.MissingBlueprintIdField;
+		};
+		const blueprints = blueprintCache.get(blueprintId) orelse {
+			std.log.err("['{s}'] Could not find blueprint '{s}'.", .{stringId, blueprintId});
+			return error.MissingBlueprint;
+		};
+		const self = StructureBuildingBlock{
+			.children = arenaAllocator.alloc(AliasTable(Child), childBlockStringId.items.len),
+			.blueprints = blueprints,
+		};
+		const childrenZon = zon.getChild("children");
+		for(childBlockStringId.items, 0..) |colorName, colorIndex| {
+			self.children[colorIndex] = try initChildTableFromZon(stringId, colorName, colorIndex, childrenZon.getChild(colorName));
+		}
+		return self;
+	}
+	pub fn getBlueprint(self: StructureBuildingBlock, rotation: Degrees) *BlueprintEntry {
+		return &self.blueprints[@intFromEnum(rotation)];
+	}
+	pub fn pickChild(self: StructureBuildingBlock, block: BlueprintEntry.StructureBlock, seed: *u64) *const StructureBuildingBlock {
+		return self.children[block.index].sample(seed).structure;
+	}
+};
+
+fn initChildTableFromZon(parentId: []const u8, colorName: []const u8, colorIndex: usize, zon: ZonElement) !AliasTable(Child) {
+	if(zon == .null) return .init(arenaAllocator, &.{});
+	if(zon != .array) {
+		std.log.err("['{s}'->'{s}'] Incorrect child data structure, array expected.", .{parentId, colorName});
+		return .init(arenaAllocator, &.{});
+	}
+	if(zon.array.items.len == 0) {
+		std.log.warn("['{s}'->'{s}'] Empty children list.", .{parentId, colorName});
```

Comment:
Should this be an error? Can you think of any cases where this wouldn't be a mistake?

Reply:
Technically this is valid configuration, there is nothing inherently bad about writing `.red = .{},` although it is not necessary. We can explicitly disallow it.

Reply:
Hmm, I guess it could be used to explicitly say that there are no children for that color.

Reply:
That would also allow us to log an error if there are any colors missing instead (but maybe not do that in this PR).

Reply:
Missing color configuration is valid way to eg. create tree with less branches from single stem with multiple child spawn blocks`

Reply:
Although, I think it won't work right now, as `pickChild` must return a value. I think it will be best to revisit that in the future.

Reply:
Could you please add a todo or open an issue for it then, so we don't forget?

Reply:
https://github.com/PixelGuys/Cubyz/issues/1250